python/semantic_kernel/connectors/ai/mistral_ai/prompt_execution_settings/mistral_ai_prompt_execution_settings.py

```diff
@@ -0,0 +1,34 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import logging
+from typing import Any, Literal
+
+from pydantic import Field
+
+from semantic_kernel.connectors.ai.function_call_behavior import FunctionCallBehavior
+from semantic_kernel.connectors.ai.prompt_execution_settings import PromptExecutionSettings
+
+logger = logging.getLogger(__name__)
+
+
+class MistralAIPromptExecutionSettings(PromptExecutionSettings):
+    """Common request settings for MistralAI services."""
+
+    ai_model_id: str | None = Field(None, serialization_alias="model")
+    max_tokens: int | None = Field(None, gt=0)
+    seed: int | None = None
+    temperature: float | None = Field(None, ge=0.0, le=2.0)
+    top_p: float | None = Field(None, ge=0.0, le=1.0)
+    user: str | None = None
+
+
+class MistralAIChatPromptExecutionSettings(MistralAIPromptExecutionSettings):
+    """Specific settings for the Chat Completion endpoint."""
+
+    response_format: dict[Literal["type"], Literal["text", "json_object"]] | None = None
+    tools: list[dict[str, Any]] | None = Field(None, max_length=64)
+    tool_choice: str | None = None
+    messages: list[dict[str, Any]] | None = None
+    safe_mode: bool = False
+    safe_prompt: bool = False
+    function_call_behavior: FunctionCallBehavior | None = Field(None, exclude=True)
```

Comment:
We're very close to introducing a new function calling abstraction (the PR is [here](https://github.com/microsoft/semantic-kernel/pull/6910)) that will allow us to better work with function calling and non-OpenAI models. Depending upon when this new connector code lands, we'll want to make sure it works with `FunctionChoiceBehavior` instead of with the legacy `FunctionCallBehavior`. Given that Mistral's tool_choice slightly differs from our base implementation, we'll need a way to map from the one difference: `any` -> `required`.

Reply:
Thats good to know, so i will probably wait integrating with Function Calling till this one is closed.

Does it makes sense, to probably split the work into 3 Pull Requests :
1. ChatCompletion Streaming & Non Streaming
2. Embbedding Integration
3. Function Calling Integration with new Abstraction

Otherwise the Request will be very big, i think it's very big already.

Reply:
That seems reasonable, yes.

Reply:
I migrated my code to the new FunctionChoiceBehavior